colourblind8/colourblind8.py

```diff
@@ -1,61 +1,59 @@
-
-    def set_prota(self,plot):
-       '''
-        A function that converts a matplotlib plot to a Protanopia friendly palette
-        and theme with more readable font size and theme design
-
-        inputs
-        ------
-        plot:  a  matplotlib figure that needs reformatting
-
-        outputs:
-        --------
-        new_plot: this is a plot that has been transformed to a Protanopia
-                  vision friendly graph, with changes to the background,
-                  gridlines, and colour
-
-        '''
-        return new_plot
-
-       def set_trita(self,plot):
-       '''
-        A function that converts a matplotlib plot to a Tritanopia friendly palette
-        and theme with more readable font size and theme design
-
-        inputs
-        ------
-        plot: a  matplotlib figure that needs reformatting
-
-        outputs:
-        --------
-        new_plot: this is a plot that has been transformed to a Tritanopia
-                  vision friendly graph, with changes to the background,
-                  gridlines, and colour
-
-        '''
-        return new_plot
-
 
+    def __init__(self):
+        self.deutera_colours =  ['#666E95', '#FFC13D', '#2F86E5', '#ACA6C3', '#483E3C',
+                                 '#E8C6BA', '#867065', '#005186', '#B0860F', '#FFD9B0']
+
+
+        self.prota_colours = ['#7D7C01', '#59709E', '#4C5631', '#3F59E8', '#BDBB64',
+                              '#35A9E0', '#E8E602', '#C1C1C7', '#0E1079']
+
+        self.trita_colours = ['#5F727A', '#FFB7C2', '#01919A', '#A8A8B4', '#BDE6F4',
+                              '#4A3F45', '#B77B87', '#00585C', '#932929']
+
+        plt.rcParams['axes.facecolor']= "white"
+        plt.rcParams['axes.titlesize'] = 20
+        plt.rcParams['axes.edgecolor'] = "black"
+        plt.rcParams['legend.fontsize'] = 10
+        plt.rcParams["legend.facecolor"] = "white"
+        plt.rcParams["figure.facecolor"] = "white"
+        plt.rcParams["axes.labelsize"] = 16
+
+    def plot_scatter(self, x, y, labels = None, palette = None,
+                title = None, x_lab = None, y_lab = None, legend_title = None):
+
+        # Checks colour palette
+        if palette == 'deutera':
+                colours = self.deutera_colours
+        elif palette == 'trita':
+                colours = self.trita_colours
+        elif palette == 'prota':
+                colours = self.prota_colours
+
+        fig, ax = plt.subplots()
+
+        if labels:
+            for idx, values in enumerate(y):
+                ax.scatter(x,
+                          values,
+                          edgecolor = 'white',
+                          color = colours[idx],
+                          label = labels[idx])
+        else:
+            for idx, values in enumerate(y):
+                ax.scatter(x,
+                           values,
+                           edgecolor = 'white',
+                           color = colours[idx])
+
+        if title:
```

Comment:
Do you think there is a way to add this to the class instead of each function? 
Like we do title parameter and x_lab parameters for the class instead of the method? 
What do you think?

Reply:
I don't agree, each function call will have it's own title. It's not data being stored in the class

Reply:
Ok I accept this.